service/src/test/java/org/databiosphere/workspacedataservice/pact/TDRPactTest.java

```diff
@@ -0,0 +1,125 @@
+import bio.terra.datarepo.model.SnapshotModel;
+import java.util.Map;
+import java.util.UUID;
+import org.databiosphere.workspacedataservice.datarepo.DataRepoClientFactory;
+import org.databiosphere.workspacedataservice.datarepo.DataRepoDao;
+import org.databiosphere.workspacedataservice.datarepo.DataRepoException;
+import org.databiosphere.workspacedataservice.datarepo.HttpDataRepoClientFactory;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+@Tag("pact-test")
+@ExtendWith(PactConsumerTestExt.class)
+class TDRPactTest {
+
+  @BeforeEach
+  void setUp() {
+    // Without this setup, the HttpClient throws a "No thread-bound request found" error
+    MockHttpServletRequest request = new MockHttpServletRequest();
+    // Set the mock request as the current request context
+    RequestContextHolder.setRequestAttributes(new ServletRequestAttributes(request));
+  }
+
+  static final String dummySnapshotId = "12345678-abc9-012d-3456-e7fab89cd01e";
+
+  @Pact(consumer = "wds", provider = "datarepo")
+  public RequestResponsePact noSnapshotPact(PactDslWithProvider builder) {
+    return builder
+        .given("snapshot doesn't exist", Map.of("id", dummySnapshotId))
+        .uponReceiving("a snapshot request")
+        .path("/api/repository/v1/snapshots/" + dummySnapshotId)
+        .query("include=TABLES")
+        .method("GET")
+        .willRespondWith()
+        .status(404)
+        .toPact();
+  }
+
+  @Pact(consumer = "wds", provider = "datarepo")
+  public RequestResponsePact noAccessToSnapshotPact(PactDslWithProvider builder) {
+    return builder
+        .given("user does not have access to snapshot")
+        .uponReceiving("a snapshot request")
+        .path("/api/repository/v1/snapshots/" + dummySnapshotId)
+        .query("include=TABLES")
+        .method("GET")
+        .willRespondWith()
+        .status(403)
+        .toPact();
+  }
+
+  @Pact(consumer = "wds", provider = "datarepo")
+  public RequestResponsePact userHasAccessToSnapshotPact(PactDslWithProvider builder) {
+    // Current use of datarepo snapshot only relies on id and name
+    // Future development should update expected response shape to include fields we need
+    var snapshotResponseShape =
+        new PactDslJsonBody().stringValue("id", dummySnapshotId).stringType("name");
+    return builder
+        .given("user has access to snapshot")
+        .uponReceiving("a snapshot request")
+        .path("/api/repository/v1/snapshots/" + dummySnapshotId)
+        .query("include=TABLES")
+        .method("GET")
+        .willRespondWith()
+        .status(200)
+        .body(snapshotResponseShape)
```

Comment:
For a simple response shape like this, inlining the definition and using the fancy new DSL can make the contract more readable.  You also get to avoid coming up with a name like `snapshotResponseShape`!

```suggestion
        .body(newJsonBody(snapshot -> {
           // toString() only needed if you apply my suggestion to strongly type dummyId
           snapshot.stringValue("id", dummySnapshotId.toString());
           snapshot.stringType("name");
        }).build())
```